### ISSUE TITLE
ERROR:  type "string" does not exist

### ISSUE THREAD

**Issue by goravbhootra** [makandra/active_type#19]
_PG::UndefinedObject - ERROR:  type "string" does not exist_

Anybody getting the above error should use :varchar or :text instead of :string. Eg.

    class SignIn < ActiveType::Object
        attribute :email_id, :varchar
        attribute :password, :varchar
        attribute :remember_me, :boolean

Basically you need to stick to DBMS' datatype. Since postgres doesn't understand "string", it throws error.

**Comment by thomasklemm** [comment]
We're using `active_type` with Postgres and haven't seen these errors yet. Which versions of `rails` and the `pg` gems are you using?

**Comment by goravbhootra** [comment]
Rails 4.2.0 with pg 0.17.1

**Comment by maxd** [comment]
The same problem in same environment. Here is backtrace:

~~~
application/vendor/bundle/ruby/2.1.0/gems/activerecord-4.2.0/lib/active_record/connection_adapters/postgresql/database_statements.rb:155:in `async_exec': PG::UndefinedObject: ERROR:  type "string" does not exist (ActiveRecord::StatementInvalid)

LINE 1: SELECT 'string'::regtype::oid
               ^
: SELECT 'string'::regtype::oid
  from application/vendor/bundle/ruby/2.1.0/gems/activerecord-4.2.0/lib/active_record/connection_adapters/postgresql/database_statements.rb:155:in `block in execute'
  from application/vendor/bundle/ruby/2.1.0/gems/activerecord-4.2.0/lib/active_record/connection_adapters/abstract_adapter.rb:466:in `block in log'
  from application/vendor/bundle/ruby/2.1.0/gems/activesupport-4.2.0/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
  from application/vendor/bundle/ruby/2.1.0/gems/activerecord-4.2.0/lib/active_record/connection_adapters/abstract_adapter.rb:460:in `log'
  from application/vendor/bundle/ruby/2.1.0/gems/activerecord-4.2.0/lib/active_record/connection_adapters/postgresql/database_statements.rb:154:in `execute'
  from application/vendor/bundle/ruby/2.1.0/gems/activerecord-4.2.0/lib/active_record/connection_adapters/postgresql_adapter.rb:393:in `lookup_cast_type'
  from application/vendor/bundle/ruby/2.1.0/gems/active_type-0.3.1/lib/active_type/type_caster.rb:68:in `initialize'
  from application/vendor/bundle/ruby/2.1.0/gems/active_type-0.3.1/lib/active_type/type_caster.rb:8:in `new'
  from application/vendor/bundle/ruby/2.1.0/gems/active_type-0.3.1/lib/active_type/type_caster.rb:8:in `get'
  from application/vendor/bundle/ruby/2.1.0/gems/active_type-0.3.1/lib/active_type/virtual_attributes.rb:48:in `add_virtual_column'
  from application/vendor/bundle/ruby/2.1.0/gems/active_type-0.3.1/lib/active_type/virtual_attributes.rb:40:in `build'
  from application/vendor/bundle/ruby/2.1.0/gems/active_type-0.3.1/lib/active_type/virtual_attributes.rb:197:in `attribute'
  from application/app/models/streaming/vsp_video.rb:4:in `<class:VspVideo>'
  ...
~~~

**Comment by bodrovis** [comment]
Same here, Rails 4.2.0, pg 0.18.1.

**Comment by shlima** [comment]
#23

**Comment by mattbrictson** [comment]
I'm also seeing this error with Rails 4.2.0, pg 0.18.1.

This fails:

```ruby
attribute :first_name, :string
```

Removing the type makes it work:

```ruby
attribute :first_name
```

**Comment by kratob** [comment]
This should be resolved with the 0.3.2 release.
